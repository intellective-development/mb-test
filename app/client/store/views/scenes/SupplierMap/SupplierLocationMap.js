// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import { withGoogleMap, GoogleMap, Marker } from 'react-google-maps';
import { address_helpers } from 'store/business/address';
import { delivery_method_constants } from 'store/business/delivery_method';
import { supplier_helpers } from 'store/business/supplier';
import type { Supplier } from 'store/business/supplier';

import { MBText } from '../../elements';

const GOOGLE_MAP_OPTIONS = {
  backgroundColor: '#fff',
  clickableIcons: false,
  disableDefaultUI: true,
  maxZoom: 20,
  minZoom: 10,
  scaleControl: true,
  tilt: 0
};

type SupplierLocationMapProps = {supplier: Supplier};
const SupplierLocationMap = ({supplier}: SupplierLocationMapProps) => {
  if (!supplier) return null;

  return (
    <div>
      <SupplierLocation supplier={supplier} />
      <SupplierMap
        supplier={supplier}
        containerElement={<div className="sm__map__container" />}
        mapElement={<div className="sm__map__container" />} />
    </div>
  );
};

const SupplierLocation = ({supplier}) => {
  const has_pickup = supplier.delivery_methods.some(dm => dm.type === delivery_method_constants.PICKUP);

  return (
    <div className="sm__header__container">
      <MBText.H4 className="sm__header__name">{supplier.name}</MBText.H4>
      <MBText.H5 className="sm__header__location">
        {address_helpers.formatStreetAndCity(supplier.address)}
        <SupplierDistance supplier={supplier} is_hidden={!has_pickup} />
      </MBText.H5>
    </div>
  );
};

const SupplierDistance = ({supplier, is_hidden}) => {
  if (is_hidden) return null;

  return (
    <MBText.Span>
      ãƒ»{supplier_helpers.formatDistance(supplier)}
    </MBText.Span>
  );
};

const SupplierMap = withGoogleMap(({supplier}) => {
  const supplier_lat_lng = new google.maps.LatLng(supplier.address.latitude, supplier.address.longitude);

  return (
    <GoogleMap
      defaultZoom={17}
      defaultCenter={supplier_lat_lng}
      options={GOOGLE_MAP_OPTIONS}>
      <Marker position={supplier_lat_lng} />
    </GoogleMap>
  );
});

const SupplierLocationMapSTP = () => {
  const findSupplier = Ent.query(Ent.find('supplier'), Ent.join('delivery_methods'));

  return (state, {supplier_id}) => ({
    supplier: findSupplier(state, supplier_id)
  });
};
const SupplierLocationMapContainer = connect(SupplierLocationMapSTP)(SupplierLocationMap);

export default SupplierLocationMapContainer;
