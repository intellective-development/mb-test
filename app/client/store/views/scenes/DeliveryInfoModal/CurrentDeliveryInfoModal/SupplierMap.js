// @flow

import * as React from 'react';

import { MBModal } from '../../../elements';
import SupplierLocationMap from '../../SupplierMap/SupplierLocationMap';

type SupplierMapProps = {supplier_id: number, routeBack: Function, hideModal: Function};
const SupplierMap = ({supplier_id, routeBack, hideModal}: SupplierMapProps) => {
  return (
    <div>
      <MBModal.SectionHeader
        renderLeft={() => <MBModal.Back onClick={routeBack} />}
        renderRight={() => <MBModal.Close onClick={hideModal} />} />
      <SupplierLocationMap supplier_id={supplier_id} />
    </div>
  );
};

export default SupplierMap;
