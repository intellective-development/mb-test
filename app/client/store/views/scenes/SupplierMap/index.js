// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import { ui_actions, ui_selectors } from 'store/business/ui';

import { MBModal } from '../../elements';
import SupplierLocationMap from './SupplierLocationMap';

type SupplierMapModalProps = {show: number, supplier_id: number, dismiss: Function};
export const SupplierMapModal = ({show, supplier_id, dismiss}: SupplierMapModalProps) => {
  return (
    <MBModal.Modal size="large" show={show} onHide={dismiss} >
      <MBModal.SectionHeader
        renderRight={() => <MBModal.Close onClick={dismiss} />} />
      <SupplierLocationMap supplier_id={supplier_id} />
    </MBModal.Modal>
  );
};

const SupplierMapModalSTP = (state) => ({
  show: ui_selectors.showSupplierMap(state),
  supplier_id: ui_selectors.mapSupplierId(state)
});
const SupplierMapModalDTP = {dismiss: ui_actions.hideSupplierMapModal};
const SupplierMapModalContainer = connect(SupplierMapModalSTP, SupplierMapModalDTP)(SupplierMapModal);

export default SupplierMapModalContainer;
