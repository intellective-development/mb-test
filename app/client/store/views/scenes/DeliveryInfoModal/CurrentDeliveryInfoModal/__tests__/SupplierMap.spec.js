import * as React from 'react';
import supplier_factory from '@minibar/store-business/src/supplier/__tests__/supplier.factory';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import SupplierMap from '../SupplierMap';

describe('SupplierMap', () => {
  it('renders', () => {
    const supplier_id = 1;
    const supplier = supplier_factory.build('with_delivery_methods', {id: supplier_id});
    const state = supplier_factory.stateify([supplier]);

    expect(render(
      <TestProvider initial_state={state}>
        <SupplierMap supplier_id={supplier_id} />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
