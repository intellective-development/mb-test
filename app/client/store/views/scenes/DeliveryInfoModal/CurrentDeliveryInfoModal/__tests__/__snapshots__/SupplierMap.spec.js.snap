// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierMap renders 1`] = `
<div>
  <div
    class="elMBModal_SectionHeader elMBModal_SectionHeader_Top"
  >
    <div
      aria-label="Back"
      class="elMBModal_Back"
    >
      <img
        alt="back"
        class="el-mbicon--back"
        src="/assets/components/elements/mb-icon/back.png"
        srcset="/assets/components/elements/mb-icon/back@2x.png 2x, /assets/components/elements/mb-icon/back@3x.png 3x"
      />
      <span
        class="elMBText elMBText_ResetSpacing elMBModal_BackText"
      >
        Â Back
      </span>
    </div>
    <h4
      class="elMBText elMBText_ResetSpacing elMBModal_SectionHeaderText"
    />
    <div
      aria-label="Close"
      class="elMBModal_Close"
    >
      <img
        alt="close"
        class="el-mbicon--close"
        src="/assets/components/elements/mb-icon/close.png"
        srcset="/assets/components/elements/mb-icon/close@2x.png 2x, /assets/components/elements/mb-icon/close@3x.png 3x"
      />
    </div>
  </div>
  <div>
    <div
      class="sm__header__container"
    >
      <h4
        class="elMBText elMBText_ResetSpacing sm__header__name"
      >
        Manhattan Wine
      </h4>
      <h5
        class="elMBText elMBText_ResetSpacing sm__header__location"
      >
         123 Jump Street, New York, New York
      </h5>
    </div>
    <div />
  </div>
</div>
`;
